scene/2d/physics/collision_shape_2d.cpp

```diff
@@ -208,6 +208,13 @@ bool CollisionShape2D::is_disabled() const {
 	return disabled;
 }
 
+bool CollisionShape2D::contains_point(const Vector2 &p_point) const {
+	if (!collision_object || !Object::cast_to<Node2D>(collision_object->shape_owner_get_owner(owner_id))) {
+		return false;
+	}
+	return shape->contains_point(to_local(p_point));
```

Comment:
The math I was doing prior to using to_local would work, given its not node-cached. The validation was originally there for that math. 

Personally speaking, running point-checking off the current physics state makes more sense given what it would be used for during runtime, and is more imminently useful. Seems overall better for the end-user.